build.cake

```diff
@@ -240,6 +243,43 @@ private void UpdateiOSEntitlementsPlist(string entitlementsPath, VariantConfig b
     Information($"{entitlementsPath} updated with success!");
 }
 
+private void UpdateWatchKitAppInfoPlist(string plistPath, VariantConfig buildVariant)
+{
+    var plistFile = File(plistPath);
+    dynamic plist = DeserializePlist(plistFile);
+
+    var prevBundleId = plist["NSExtension"]["NSExtensionAttributes"]["WKAppBundleIdentifier"];
+    var newBundleId = GetiOSBundleId(buildVariant, iOSProjectType.WatchApp);
+
+    plist["NSExtension"]["NSExtensionAttributes"]["WKAppBundleIdentifier"] = newBundleId;
+
+    SerializePlist(plistFile, plist);
+
+    Information($"Changed Bundle Identifier from {prevBundleId} to {newBundleId}");
+    Information($"{plistPath} updated with success!");
+}
+
+private void UpdateWatchPbxproj(string pbxprojPath, GitVersion git)
+{
+    var fileText = FileReadText(pbxprojPath);
+    if (string.IsNullOrEmpty(fileText))
+    {
+        throw new Exception($"Couldn't find {pbxprojPath}");
+    }
+
+    const string pattern = @"MARKETING_VERSION = [^;]*;";
+
+    var plistFile = File(Path.Combine(_slnPath, "src", "iOS", "Info.plist"));
+    dynamic plist = DeserializePlist(plistFile);
```

Comment:
From what I'm understanding we're only deserializing the iOS plist here to fetch the previously set version string. If that's the case, we can instead:

1. In the `UpdateiOSPlist` task, save that string in a top level var. You'll find them at the top of the script, their names start with `_` and have a comment explaining where they're set
2. Use a [Criteria](https://cakebuild.net/docs/writing-builds/tasks/criteria) in the `UpdateWatchProject` to check that var is filled. This will mostly serve as a way to raise visibility of the dependency from another task. E.g. `.WithCriteria(!string.IsNullOrEmpty(_iOSVersionName))`